lib/nvme/nvme_ctrlr.c

```diff
@@ -849,6 +841,9 @@ nvme_ctrlr_process_init(struct spdk_nvme_ctrlr *ctrlr)
 			cc.bits.en = 0;
 			nvme_mmio_write_4(ctrlr, cc.raw, cc.raw);
 			nvme_ctrlr_set_state(ctrlr, NVME_CTRLR_STATE_DISABLE_WAIT_FOR_READY_0, ready_timeout_in_ms);
+
+			if (ctrlr->quirks & NVME_QUIRK_DELAY_BEFORE_CHK_RDY)
+				sleep(2);
```

Comment:
Adding a sleep() here will block any other controllers currently being initialized in this spdk_nvme_probe() call.  In the worst case, if all of the initializing devices require this quirk, the total delay will be 2 seconds times the number of devices, rather than just 2 seconds waiting for all devices in parallel.

Would it be possible to add a new state to NVME_CTRLR_STATE and use nvme_ctrlr_set_state() to wait for the correct amount of time?  That way, other controllers could continue through the state machine.